<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>rncbc aka Rui Nuno Capela</author>
 <comment>qtractor - An Audio/MIDI multi-track sequencer.

   Copyright (C) 2005-2024, rncbc aka Rui Nuno Capela. All rights reserved.

   This program is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public License
   as published by the Free Software Foundation; either version 2
   of the License, or (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License along
   with this program; if not, write to the Free Software Foundation, Inc.,
   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

 </comment>
 <class>qtractorMidiControlForm</class>
 <widget class="QDialog" name="qtractorMidiControlForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Controllers</string>
  </property>
  <property name="windowIcon">
   <iconset theme="itemControllers" />
  </property>
  <layout class="QGridLayout">
   <item row="0" column="0" rowspan="4" colspan="9">
    <widget class="QTreeWidget" name="FilesListView">
     <property name="toolTip">
      <string>Controller files</string>
     </property>
     <property name="rootIsDecorated">
      <bool>false</bool>
     </property>
     <property name="uniformRowHeights">
      <bool>true</bool>
     </property>
     <property name="itemsExpandable">
      <bool>false</bool>
     </property>
     <property name="allColumnsShowFocus">
      <bool>true</bool>
     </property>
     <column>
      <property name="text">
       <string>Files</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Path</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="0" column="9" colspan="3">
    <widget class="QPushButton" name="ImportPushButton">
     <property name="toolTip">
      <string>Import controller files</string>
     </property>
     <property name="text">
      <string>&amp;Import...</string>
     </property>
     <property name="icon">
      <iconset theme="formOpen" />
     </property>
    </widget>
   </item>
   <item row="1" column="9" colspan="3">
    <widget class="QPushButton" name="RemovePushButton">
     <property name="toolTip">
      <string>Remove controller file</string>
     </property>
     <property name="text">
      <string>&amp;Remove</string>
     </property>
     <property name="icon">
      <iconset theme="formRemove" />
     </property>
    </widget>
   </item>
   <item row="2" column="9" colspan="3">
    <widget class="QPushButton" name="MoveUpPushButton">
     <property name="toolTip">
      <string>Move controller file up on list order</string>
     </property>
     <property name="text">
      <string>&amp;Up</string>
     </property>
     <property name="icon">
      <iconset theme="formMoveUp" />
     </property>
    </widget>
   </item>
   <item row="3" column="9" colspan="3">
    <widget class="QPushButton" name="MoveDownPushButton">
     <property name="toolTip">
      <string>Move controller file down on list order</string>
     </property>
     <property name="text">
      <string>&amp;Down</string>
     </property>
     <property name="icon">
      <iconset theme="formMoveDown" />
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="ControlTypeTextLabel">
     <property name="text">
      <string>&amp;Type</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="buddy">
      <cstring>ControlTypeComboBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QLabel" name="ChannelTextLabel">
     <property name="text">
      <string>&amp;Channel</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="buddy">
      <cstring>ChannelComboBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="2">
    <widget class="QLabel" name="ParamTextLabel">
     <property name="text">
      <string>&amp;Parameter</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="buddy">
      <cstring>ParamComboBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="3">
    <widget class="QLabel" name="TrackParamTextLabel">
     <property name="text">
      <string>Trac&amp;k</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="buddy">
      <cstring>TrackOffsetSpinBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="4">
    <widget class="QLabel" name="TrackOffsetTextLabel">
     <property name="text">
      <string>offse&amp;t</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="buddy">
      <cstring>TrackOffsetSpinBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="5">
    <widget class="QLabel" name="TrackLimitTextLabel">
     <property name="text">
      <string>&amp;limit</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="buddy">
      <cstring>TrackLimitSpinBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="6">
    <widget class="QLabel" name="CommandTextLabel">
     <property name="text">
      <string>C&amp;ommand</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
     <property name="buddy">
      <cstring>CommandComboBox</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="7" colspan="2">
    <widget class="QLabel" name="FlagsTextLabel">
     <property name="text">
      <string>Flags</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
     </property>
    </widget>
   </item>
   <item row="4" column="9" colspan="3">
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint">
      <size>
       <width>20</width>
       <height>8</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="5" column="0">
    <widget class="QComboBox" name="ControlTypeComboBox">
     <property name="toolTip">
      <string>MIDI Event type</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QComboBox" name="ChannelComboBox">
     <property name="toolTip">
      <string>MIDI Channel</string>
     </property>
    </widget>
   </item>
   <item row="5" column="2">
    <widget class="QComboBox" name="ParamComboBox">
     <property name="toolTip">
      <string>MIDI Controller (parameter)</string>
     </property>
    </widget>
   </item>
   <item row="5" column="3">
    <widget class="QCheckBox" name="TrackParamCheckBox">
     <property name="toolTip">
      <string>MIDI parameter (track offset)</string>
     </property>
     <property name="text">
      <string>+</string>
     </property>
    </widget>
   </item>
   <item row="5" column="4">
    <widget class="QSpinBox" name="TrackOffsetSpinBox">
     <property name="toolTip">
      <string>Track offset</string>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>127</number>
     </property>
    </widget>
   </item>
   <item row="5" column="5">
    <widget class="QSpinBox" name="TrackLimitSpinBox">
     <property name="toolTip">
      <string>Track limit</string>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>127</number>
     </property>
    </widget>
   </item>
   <item row="5" column="6">
    <widget class="QComboBox" name="CommandComboBox">
     <property name="toolTip">
      <string>Command action</string>
     </property>
    </widget>
   </item>
   <item row="5" column="7">
    <widget class="QCheckBox" name="DeltaCheckBox">
     <property name="toolTip">
      <string>Command delta/momentary</string>
     </property>
     <property name="text">
      <string>D&amp;elta</string>
     </property>
    </widget>
   </item>
   <item row="5" column="8">
    <widget class="QCheckBox" name="FeedbackCheckBox">
     <property name="toolTip">
      <string>Command feedback</string>
     </property>
     <property name="text">
      <string>&amp;Feedback</string>
     </property>
    </widget>
   </item>
   <item row="5" column="9" colspan="3">
    <widget class="QPushButton" name="MapPushButton">
     <property name="toolTip">
      <string>Map/update controller command</string>
     </property>
     <property name="text">
      <string>&amp;Map</string>
     </property>
     <property name="icon">
      <iconset theme="formAccept" />
     </property>
    </widget>
   </item>
   <item row="6" column="0" rowspan="7" colspan="9">
    <widget class="QTreeWidget" name="ControlMapListView">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>160</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Controller map</string>
     </property>
     <property name="rootIsDecorated">
      <bool>false</bool>
     </property>
     <property name="uniformRowHeights">
      <bool>true</bool>
     </property>
     <property name="itemsExpandable">
      <bool>false</bool>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <property name="allColumnsShowFocus">
      <bool>true</bool>
     </property>
     <column>
      <property name="text">
       <string>Type</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Channel</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Parameter</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Track</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Command</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Flags</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="6" column="9" colspan="3">
    <widget class="QPushButton" name="UnmapPushButton">
     <property name="toolTip">
      <string>Unmap/remove controller command</string>
     </property>
     <property name="text">
      <string>U&amp;nmap</string>
     </property>
     <property name="icon">
      <iconset theme="formReject" />
     </property>
    </widget>
   </item>
   <item row="7" column="9" colspan="3">
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint">
      <size>
       <width>20</width>
       <height>8</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="8" column="10">
    <widget class="QCheckBox" name="SyncCheckBox">
     <property name="toolTip">
      <string>Enable all controllers immediate sync (hook)</string>
     </property>
     <property name="text">
      <string>&amp;Sync</string>
     </property>
    </widget>
   </item>
   <item row="9" column="9" colspan="3">
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint">
      <size>
       <width>20</width>
       <height>8</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="10" column="9" colspan="3">
    <widget class="QPushButton" name="ReloadPushButton">
     <property name="toolTip">
      <string>Reload/apply all controller files</string>
     </property>
     <property name="text">
      <string>Relo&amp;ad</string>
     </property>
     <property name="icon">
      <iconset theme="formRefresh" />
     </property>
    </widget>
   </item>
   <item row="11" column="9" colspan="3">
    <widget class="QPushButton" name="ExportPushButton">
     <property name="toolTip">
      <string>Export to controller file</string>
     </property>
     <property name="text">
      <string>E&amp;xport...</string>
     </property>
     <property name="icon">
      <iconset theme="formSave" />
     </property>
    </widget>
   </item>
   <item row="12" column="9" colspan="3">
    <widget class="QPushButton" name="ClosePushButton">
     <property name="toolTip">
      <string>Close this dialog</string>
     </property>
     <property name="text">
      <string>Close</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="4" margin="8"/>
 <tabstops>
  <tabstop>FilesListView</tabstop>
  <tabstop>ImportPushButton</tabstop>
  <tabstop>RemovePushButton</tabstop>
  <tabstop>MoveUpPushButton</tabstop>
  <tabstop>MoveDownPushButton</tabstop>
  <tabstop>ControlTypeComboBox</tabstop>
  <tabstop>ChannelComboBox</tabstop>
  <tabstop>ParamComboBox</tabstop>
  <tabstop>TrackParamCheckBox</tabstop>
  <tabstop>TrackOffsetSpinBox</tabstop>
  <tabstop>TrackLimitSpinBox</tabstop>
  <tabstop>CommandComboBox</tabstop>
  <tabstop>DeltaCheckBox</tabstop>
  <tabstop>FeedbackCheckBox</tabstop>
  <tabstop>MapPushButton</tabstop>
  <tabstop>ControlMapListView</tabstop>
  <tabstop>UnmapPushButton</tabstop>
  <tabstop>SyncCheckBox</tabstop>
  <tabstop>ReloadPushButton</tabstop>
  <tabstop>ExportPushButton</tabstop>
  <tabstop>ClosePushButton</tabstop>
 </tabstops>
 <resources>
  <include location="qtractor.qrc"/>
 </resources>
 <connections/>
</ui>
